tracing/src/main/java/org/axonframework/extensions/tracing/SpanUtils.java

```diff
@@ -41,24 +41,28 @@
      * <p>
      * This method will check if the message is a
      * <ul>
-     * <li>{@link QueryMessage}, returning "{@code QueryMessage}",</li>
-     * <li>{@link CommandMessage}, returning "{@code CommandMessage}",</li>
-     * <li>{@link EventMessage}, returning "{@code EventMessage}",</li>
+     * <li>{@link QueryMessage}, returning  the queryName",</li>
+     * <li>{@link CommandMessage}, returning the commandName",</li>
+     * <li>{@link EventMessage}, returning the event type,</li>
      * <li>otherwise returns "{@code Message}"</li>
      * </ul>
      *
      * @param message The message to resolve the type of
      * @return a String describing the type of message
      */
-    public static String resolveType(Message message) {
-        Class<?> clazz = Message.class;
-        if (message instanceof QueryMessage) {
-            clazz = QueryMessage.class;
-        } else if (message instanceof CommandMessage) {
-            clazz = CommandMessage.class;
-        } else if (message instanceof EventMessage) {
-            clazz = EventMessage.class;
+    public static String messageName(Message message) {
+        if (message instanceof QueryMessage && !message.getPayloadType().getName().equals(((QueryMessage) message).getQueryName())) {
```

Comment:
hm, this would look like : 
```
  private static boolean messageHasName(Message message) {
        if (message instanceof QueryMessage) {
            return !message.getPayloadType().getName().equals(((QueryMessage) message).getQueryName());
        } else if (message instanceof CommandMessage && messageHasName(message)) {
            return !message.getPayloadType().getName().equals(((CommandMessage) message).getCommandName());
        }
        return false;;
    }
```
repeating the instanceof thing, which i want to have as less as possible anyway.  Woud be better if messageName was a method of the abstract `Message` Type then.

Reply:
made use of the other `messageName` method